src/components/vouchers/VouchersList.tsx

```diff
@@ -0,0 +1,119 @@
+import { StackScreenProps } from '@react-navigation/stack';
+import { FlashList } from '@shopify/flash-list';
+import React, { useEffect, useState } from 'react';
+
+import { useRenderItem } from '../../hooks';
+import { QUERY_TYPES } from '../../queries';
+import { TVoucherItem } from '../../types';
+import { LoadingSpinner } from '../LoadingSpinner';
+
+const keyExtractor = (item: TVoucherItem, index: number) => `index${index}-id${item.id}`;
+
+const MAX_INITIAL_NUM_TO_RENDER = 15;
+
+const sectionData = (data: TVoucherItem[]) => {
+  const groupDataByCategory = (data) => {
+    const grouped = {};
+    data.forEach((item: TVoucherItem) => {
+      if (item.categories?.length) {
+        if (!grouped[item.categories[0].name]) {
+          grouped[item.categories[0].name] = [];
+        }
+        grouped[item.categories[0].name].push(item);
+      }
+    });
+    return grouped;
+  };
+
+  const transformGroupedDataToArray = (groupedData) => {
+    const resultArray = [];
+    for (const category in groupedData) {
+      resultArray.push(category);
+      resultArray.push(...groupedData[category]);
+    }
+    return resultArray;
```

Comment:
please insert line breaks after `const`s and before `return`s for readability

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/988/commits/de161547c3488ae2a87cbe6a5163c6cea10564e2